discovery-provider/src/tasks/index_user_bank.py

```diff
@@ -0,0 +1,348 @@
+
+                # Restart processing at the end of this transaction signature batch
+                last_tx = transactions_array[-1]
+                last_tx_signature = last_tx["signature"]
+
+                # Append batch of processed signatures
+                if transaction_signature_batch:
+                    transaction_signatures.append(transaction_signature_batch)
+
+                # Ensure processing does not grow unbounded
+                if len(transaction_signatures) > TX_SIGNATURES_MAX_BATCHES:
+                    logger.info(f"index_user_bank.py | slicing tx_sigs from {len(transaction_signatures)} entries")
+                    transaction_signatures = transaction_signatures[-TX_SIGNATURES_RESIZE_LENGTH:]
+                    logger.info(f"index_user_bank.py | sliced tx_sigs to {len(transaction_signatures)} entries")
+
+                # Reset batch state
+                transaction_signature_batch = []
+
+    # Reverse batches aggregated so oldest transactions are processed first
+    transaction_signatures.reverse()
+
+    num_txs_processed = 0
+    for tx_sig_batch in transaction_signatures:
+        logger.error(f"index_user_bank.py | processing {tx_sig_batch}")
+        batch_start_time = time.time()
+        # Process each batch in parallel
+        with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
+            with db.scoped_session() as session:
+                parse_sol_tx_futures = {
+                    executor.submit(
+                        parse_user_bank_transaction,
+                        session,
+                        solana_client,
+                        tx_sig
+                        ): tx_sig
+                    for tx_sig in tx_sig_batch
+                }
+                for future in concurrent.futures.as_completed(parse_sol_tx_futures):
+                    try:
+                        # No return value expected here so we just ensure all futures are resolved
+                        future.result()
+                        num_txs_processed += 1
+                    except Exception as exc:
+                        logger.error(f"index_user_bank.py | error {exc}", exc_info=True)
+                        raise
+
+        batch_end_time = time.time()
+        batch_duration = batch_end_time - batch_start_time
+        logger.info(
+            f"index_user_bank.py | processed batch {len(tx_sig_batch)} txs in {batch_duration}s"
+        )
+
+
+######## CELERY TASKS ########
+@celery.task(name="index_user_bank", bind=True)
+def index_user_bank(self):
+    # Cache custom task class properties
+    # Details regarding custom task context can be found in wiki
+    # Custom Task definition can be found in src/__init__.py
+    redis = index_user_bank.redis
+    # Define lock acquired boolean
+    have_lock = False
+    # Define redis lock object
+    update_lock = redis.lock("user_bank_lock", blocking_timeout=25)
```

Comment:
IIRC, having blocking_timeout set but blocking=False on the lock acquisition doesn't make sense / do anything. worth double checking

Reply:
updated to timeout instead